api/src/resources/source.rs

```diff
@@ -22,6 +22,7 @@ pub struct Source {
     pub should_translate: bool,
     pub created_at: DateTime<Utc>,
     pub updated_at: DateTime<Utc>,
+    pub _type: String,
```

Comment:
1. The source type is one of a few cases, so could really be an `enum`. One note of caution -- in the past enums here were a maintenance burder in some places where the list would change often (as such entities or permissions). `String` is a way to make it backwards compatible -- we can add new source types and the cli won't break if it doesn't know about that enum variant. At the same time, I don't expect the list to change often, so if it was me, I'd go for the enum -- also a more challenging change :wink:. 
If not enum, at a minimum, I think we need a new type here similar to id / owner -- can check how those are implemented, you'd have to `impl FromStr for SourceType` so you can take it directly as a cmd line arg.
1. nit. While, I agree with this being called _type in the API, I am not so sure whether this should extend to client / cli as 1. this has no notion of v1. vs v2., the source here is just an abstract resource and more pragmatically, _field in Rust by convention is an unused field (i.e. it would not warn if you don't use it, e.g. as a tag or for FFI). type is a keyword though, my nit suggestion would be type_

Reply:
1. Sure, I'll give it a go.
2. I completely agree, that's why I have in the backend, I only wanted the think layer of the api serialization to read it as `_type`, wasn't sure it was possible here, but I'll give it a go!

Reply:
Actually, I think I will need help on 2.. I don't get how to do it. How do I map the json key `_type` to the `Source` object, there's no obvious construction of a `Source` object anywhere I can see.

Reply:
As far as I can tell, the cli will call the api and map the api response to a `ResponseT`, which in the case of creating a source is `CreateResponse`, which is just `source: Source`. So the mapping is directly from the api to the `Source` object, hence I don't see how I can name the type field `Source.source_type` rather than `Source._type`. Sounds to me `Source` is just a rust representation of the api json response.

Reply:
Ok, so this worked, is it standard practice?
```
    #[serde(rename = "_type")]
    pub source_type: String,
```

Reply:
On the topic of 1., leaving aside the fun challenge of learning how to do is as an `Enum`, I'm thinking it might not be the smart choice at the moment given we're just introducing this source type and it might change a few times before it settles into a stable type. Having the `create source` cli action break each time we change the api might be something we'll regret!

I'm also not convinced if philosophically the cli should be strict on types when the api is validating. It kinda isn't the cli's responsiblity. It doesn't hurt to help the user, but I think the validation on the cli should always be more relaxed than the validation at the api. It would be pretty obvious for the user when the request is made what they've done wrong anyway.

Reply:
> On the topic of 1., leaving aside the fun challenge of learning how to do is as an Enum, I'm thinking it might not be the smart choice at the moment given we're just introducing this source type and it might change a few times before it settles into a stable type. Having the create source cli action break each time we change the api might be something we'll regret!

Newtype for string as you've done is not too bad -- but will just mention the solution for permissions was an enum which is either a known variant or an unknown one explicitly. Something like

```rust
#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
#[serde(untagged)]
enum SourceType {
    #[serde(rename = "call")]
    Call,

    #[serde(rename = "chat")]
    Chat,

    Unknown(Box<str>),
}
```

I.e. it's either a known variant or `Unknown(...)` with a string, (`Box<str>` is just a way of saying the string is immutable). You can rely on `serde`'s json parsing for parsing it, e.g. something like (that's what we seem to do for permissions at least)

```rust
impl FromStr for SourceTyoe {
    type Err = Error;

    fn from_str(string: &str) -> Result<Self> {
        serde_json::de::from_str(&format!("\"{}\"", string)).map_err(|_| {
            Error::BadSourceType {
                indentifier: string.into(),
            }
        })
    }
}
```

This introduces a new error type `Error::BadSourceType` (see `api/src/error.rs`)

Reply:
That's fair @mcobzarenco , I guess my worry is more about the fact that we ship the CLI, and we'll probably change the api of this type a few times fairly soon. I don't want our clients to have to constantly update the CLI with new updates, or worse yet, have to wait for a fix, whenever we change the `_type` field in the api, especially since it's a private, and currently unstable field.

I'm more than happy to make it a robust feature of the CLI once it's stable, but I don't really think it is at the moment.

Reply:
> I don't want our users to have to constantly update the CLI with new updates, or worse yet, have to wait for a fix, whenever we change the _type field in the api, especially since it's a private, and currently unstable field.

Just for clarity, the solution above would make it forwards compatible as new `source_type`s introduced by the API would be parsed in the `Unknown` variant of the enum